realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -56,6 +56,7 @@
     private static final String INCORRECT_THREAD_MESSAGE = "Realm access from incorrect thread. Realm objects can only be accessed on the thread they were created.";
     private static final String CLOSED_REALM_MESSAGE = "This Realm instance has already been closed, making it unusable.";
     private static final String DIFFERENT_KEY_MESSAGE = "Wrong key used to decrypt Realm.";
+    private static final String CANNOT_REFRESH_INSIDE_OF_TRANSACTION_MESSAGE = "Cannot refresh inside of a transaction, the Realm is always kept up to date.";
```

Comment:
Realm are only kept up to date if they are in a Looper thread so I would truncate the message at the comma.